src/common.rs

```diff
@@ -729,6 +775,60 @@ pub(crate) fn prune_parse_output(res: Output) -> Result<(), PruneError> {
     Ok(())
 }
 
+pub(crate) fn mount_fmt_args(options: &MountOptions, common_options: &CommonOptions) -> String {
+    format!(
+        "--log-json {common_options} mount {repository_or_archive} {mountpoint}{select_paths}",
+        repository_or_archive = options.repository_or_archive,
+        mountpoint = options.mountpoint,
+        select_paths = options
+            .select_paths
+            .iter()
+            .map(|x| format!(" --pattern={}", shlex::quote(&x.to_string()),))
+            .collect::<Vec<String>>()
+            .join(" "),
+    )
+}
+
+pub(crate) fn mount_parse_output(res: Output) -> Result<(), MountError> {
+    let Some(exit_code) = res.status.code() else {
+        warn!("borg process was terminated by signal");
+        return Err(MountError::TerminatedBySignal);
+    };
+
+    let mut output = String::new();
+
+    for line in BufRead::lines(res.stderr.as_slice()) {
+        let line = line.map_err(MountError::InvalidBorgOutput)?;
+        writeln!(output, "{line}").unwrap();
+
+        trace!("borg output: {line}");
+
+        let log_msg: LoggingMessage = serde_json::from_str(&line)?;
+
+        if let LoggingMessage::LogMessage {
+            name,
+            message,
+            level_name,
+            time,
+            msg_id,
+        } = log_msg
+        {
+            log_message(level_name, time, name, message);
+
+            if let Some(msg_id) = msg_id {
+                if exit_code > 1 {
+                    return Err(MountError::UnexpectedMessageId(msg_id));
+                }
+            }
```

Comment:
We should try to find some common errors that can occur (already mounted, etc.). As this is also used for umounting, I'd also check for an eror for "no repository mounting at the given path"

Reply:
Good point. I won't have the time tonight but I'll look again on the weekend. There's certainly errors around bad sources, permission issues on the mountpoint, does the mountpoint exist, etc.


> As this is also used for umounting, I'd also check for an error for "no repository mounting at the given path"

Good call, I'll make sure that's added

Reply:
Alright, there was some code churn but I added better support for mount errors. Unfortunately Borg doesn't do a great job of reporting {u,}mount errors:

```
# No message id
{"type": "log_message", "time": 1682798547.876958, "message": "test-mount12: Mountpoint must be a writable directory", "levelname": "ERROR", "name": "borg.archiver"}

# Not even in json
$ borg umount --log-json non-existent-directory
fusermount: entry for /home/david/non-existent-directory not found in /etc/mtab
$ borg umount --log-json test-mount
fusermount: entry for /home/david/test-mount not found in /etc/mtab
```

Lemme know if these changes are too much!